### PR TITLE
Add Min cost climbing stairs dp in C code

### PR DESCRIPTION
**Pull request by Zuoxun**
Running results are as follows:

> 输入楼梯的代价列表为 [0, 1, 10, 1, 1, 1, 10, 1, 1, 10, 1]
爬完楼梯的最低代价为 6
爬完楼梯的最低代价为 6

If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    int a = cost[1], b = cost[2];
+    for (int i = 3; i <= n; i++) {
+        int tmp = b;
+        b = min(a, tmp) + cost[i];
+        a = tmp;
+    }
+    return b;
+}
+
+/*求最小值*/
```
Comment (krahets) [inline codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:42]: /* 求最小值 */

Same as other all.

File: codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    dp[2] = cost[2];
+    // 状态转移：从较小子问题逐步求解较大子问题
+    for (int i = 3; i <= n; i++) {
+        dp[i] = min(dp[i - 1], dp[i - 2]) + cost[i];
+    }
+    return dp[n];
+}
+
+/* 爬楼梯最小代价：空间优化后的动态规划 */
+int minCostClimbingStairsDPComp(int cost[]) {
```
Comment (krahets) [inline codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:29]: Add function parameter `n`. Same as others.

File: codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    if (n == 1 || n == 2)
+        return cost[n];
+    // 初始化 dp 表，用于存储子问题的解
+    int dp[n + 1];
+    // 初始状态：预设最小子问题的解
+    dp[1] = cost[1];
+    dp[2] = cost[2];
+    // 状态转移：从较小子问题逐步求解较大子问题
+    for (int i = 3; i <= n; i++) {
+        dp[i] = min(dp[i - 1], dp[i - 2]) + cost[i];
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:23]: There is no `min` declaration and implementation in the public header file of the c code. If you use it directly here, you should get an error.

File: codes/c/utils/vector.h
Comment (Gonglja) [inline codes/c/utils/vector.h:1]: Don't follow this file

File: codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c (hunk 3 above)
Comment (Zuoxun) [reply codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:23]: I've declared function min in this file.

Comment (Gonglja) [reply codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:23]: > I've declared function min in this file.

I don’t see the `min` forward declaration before using the min function.

Comment (Zuoxun) [reply codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:23]: Sorry, i didn't declare it before using it, i'll edit tomorrow.

File: codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c
[the remark this answers is not in the document]
Comment (Zuoxun) [reply codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:1]: Running results are as follows:

> 输入楼梯的代价列表为 [0, 1, 10, 1, 1, 1, 10, 1, 1, 10, 1]
爬完楼梯的最低代价为 6
爬完楼梯的最低代价为 6

[the remark this answers is not in the document]
Comment (Zuoxun) [reply codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c:1]: Cmake file is created now.

File: codes/c/utils/vector.h
Comment (Zuoxun) [reply codes/c/utils/vector.h:1]: The file vector.h is same as original version

**Review by Gonglja — COMMENTED** [review]
Can be approved @krahets

**Review by krahets — APPROVED** [review]

**Pull request merged**